pageserver/src/tenant.rs

```diff
@@ -2359,6 +2363,11 @@ impl Tenant {
             remote_client
                 .schedule_index_upload_for_metadata_update(&metadata)
                 .context("branch initial metadata upload")?;
+
+            remote_client
+                .wait_completion()
+                .await
+                .context("wait for branch initial index_part.json to upload")?;
```

Comment:
Perhaps this should be done at the caller to have uploads finish for both `bootstrap_timeline` and `branch_timeline`. Moving it.

Reply:
Moved in [2d2c5af](https://github.com/neondatabase/neon/pull/4204/commits/2d2c5af072ef1a13a35839a0b910f09feb4b7511).